### PR TITLE
fix!: Fix tag_sets to storage_stores.read_preference in mongodbatlas_federated_database_instance

### PR DESCRIPTION
**Pull request by maastha**
## Description

Update existing `tags` attribute to be `tag_sets` in storage_stores.read_preference in mongodbatlas_federated_database_instance. 
This PR fixes the structure of the attribute to be inline with the corresponding APIs.

 _Note: This PR introduces a breaking change, hence we need to remove migration tests that were added for testing migration of this resource to new Atlas SDK._

Link to any related issue(s): [INTMDB-1027](https://jira.mongodb.org/browse/INTMDB-1027)

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

## Testing
left: .tf config file, right: .tfstate file with added tag_sets
Screenshot 2023-09-01 at 11 36 08

Acceptance tests ( locally):
Screenshot 2023-09-01 at 12 22 23

Screenshot 2023-09-01 at 12 24 49

### CODE REVIEW

File: mongodbatlas/resource_mongodbatlas_federated_database_instance_migration_test.go [hunk 1]
```diff
@@ -1,158 +0,0 @@
-package mongodbatlas
```
Comment (andreaangiolillo) [inline mongodbatlas/resource_mongodbatlas_federated_database_instance_migration_test.go:1]: why did we have these tests if the resource is still using sdkv2? 🤔

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: mongodbatlas/resource_mongodbatlas_federated_database_instance_migration_test.go (hunk 1 above)
Comment (maastha) [reply mongodbatlas/resource_mongodbatlas_federated_database_instance_migration_test.go:1]: The resource isn't using SDKv2, there were some methods in the acceptance testing files which were left out which were still using SDKv2, that has been updated now with this PR.

Comment (andreaangiolillo) [reply mongodbatlas/resource_mongodbatlas_federated_database_instance_migration_test.go:1]: > The resource isn't using SDKv2

I am a bit confused here. Is the federated resource/data source using the new framework then? It does not seem to me but I may be missing something here. Looking at the import of the data source I see `/terraform-plugin-sdk/v2`

Comment (maastha) [reply mongodbatlas/resource_mongodbatlas_federated_database_instance_migration_test.go:1]: Oh I thought you were talking about new Atlas SDK. Nm.

So the resource has not been migrated to the Plugin Framework, only migrated to use the new Atlas SDK. See my comment [here](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1415#discussion_r1310297148)

The migration tests are useful for both migrations.

**Review by Zuhairahmed — APPROVED** [review]
Can we add link in PR description to INTMDB, helpful for reviewers. Otherwise LGTM. great one @maastha!

File: mongodbatlas/resource_mongodbatlas_federated_database_instance.go [hunk 2]
```diff
@@ -816,10 +859,39 @@ func newReadPreferenceField(atlasReadPreference *admin.DataLakeAtlasStoreReadPre
 		{
 			"mode":                  atlasReadPreference.GetMode(),
 			"max_staleness_seconds": atlasReadPreference.GetMaxStalenessSeconds(),
+<<<<<<< HEAD
+			"tag_sets":              flattenTagSets(atlasReadPreference.TagSets),
+=======
+>>>>>>> master
```
Comment (andreaangiolillo) [inline mongodbatlas/resource_mongodbatlas_federated_database_instance.go]: 👀 
```suggestion
```

**Comment by maastha** [comment]
Have run the tests locally against prod since cloud-dev seems to be having issues.

File: mongodbatlas/data_source_mongodbatlas_federated_database_instance.go [hunk 3]
```diff
@@ -279,18 +279,26 @@ func schemaFederatedDatabaseInstanceStoresDataSource() *schema.Schema {
 								Type:     schema.TypeInt,
 								Computed: true,
 							},
-							"tags": {
+							"tag_sets": {
```
Comment (andreaangiolillo) [inline mongodbatlas/data_source_mongodbatlas_federated_database_instance.go:282]: here

**Pull request merged**